pkg/services/ngalert/notifier/channels/slack.go

```diff
@@ -205,7 +219,38 @@ func (sn *SlackNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool,
 	if err := sendSlackRequest(request, sn.log); err != nil {
 		return false, err
 	}
-	return true, nil
+
+	// Try to upload if we have an image path but no image URL. This uploads the file
+	// immediately after the message. A bit of a hack, but it doesn't require the
+	// user to have an image host set up.
+	// TODO: how many image files should we upload? In what order? Should we
+	// assume the alerts array is already sorted?
+	// TODO: We need a refactoring so we don't do two database reads for the same data.
+	// TODO: Should we process all alerts' annotations? We can only have on image.
+	// TODO: Should we guard out-of-bounds errors here? Callers should prevent that from happening, imo
+	imgToken := getTokenFromAnnotations(alerts[0].Annotations)
+	dbContext, cancel := context.WithTimeout(ctx, ImageStoreTimeout)
+	imgData, err := sn.images.GetData(dbContext, imgToken)
+	cancel()
+	if err != nil {
+		if !errors.Is(err, ErrImagesUnavailable) {
+			// Ignore errors. Don't log "ImageUnavailable", which means the storage doesn't exist.
+			sn.log.Warn("Error reading screenshot data from ImageStore: %v", err)
+		}
+		return true, nil
+	} else {
```

Comment:
Does the `return` on 240 mean we can drop the else and move this branch out a level?